src/nyc_trees/libs/auth_backend.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+from django.contrib.auth.backends import ModelBackend
+
+from apps.core.models import User
+
+
+class EmailOrUsernameAuthentication(ModelBackend):
+    def authenticate(self, username=None, password=None):
+        user = None
+        try:
```

Comment:
Hmm... `@` and `.` _are_ [allowed in usernames](https://github.com/django/django/blob/master/django/contrib/auth/models.py#L307-L309), and we already have some users who used email addresses as their username.

It's unlikely that someone signed up with an email address as their username which is not _their_ email, but I suppose we should try and handle that somehow.